actors/builtin/miner/miner_state_test.go

```diff
@@ -0,0 +1,587 @@
+	assert.Equal(h.t, &miner.SectorPreCommitOnChainInfo{}, out)
+}
+
+func (h *minerStateHarness) assertDeletePreCommit(rt runtime.Runtime, sectorNo abi.SectorNumber) {
+	err := h.s.DeletePrecommittedSector(adt.AsStore(rt), sectorNo)
+	assert.NoError(h.t, err)
+}
+
+//
+// Type Construction Methods
+//
+
+// returns a unique SectorPreCommitOnChainInfo with each invocation with SectorNumber set to `sectorNo`.
+func (h *minerStateHarness) newSectorPreCommitOnChainInfo(sectorNo abi.SectorNumber) *miner.SectorPreCommitOnChainInfo {
+	seed := h.getSeed()
+	info := h.newSectorPreCommitInfo(sectorNo)
+	return &miner.SectorPreCommitOnChainInfo{
+		Info:             *info,
+		PreCommitDeposit: abi.NewTokenAmount(int64(seed)),
+		PreCommitEpoch:   abi.ChainEpoch(seed),
+	}
+}
+
+// returns a unique SectorOnChainInfo with each invocation with SectorNumber set to `sectorNo`.
+func (h *minerStateHarness) newSectorOnChainInfo(sectorNo abi.SectorNumber) *miner.SectorOnChainInfo {
+	seed := h.getSeed()
+	info := h.newSectorPreCommitInfo(sectorNo)
+	return &miner.SectorOnChainInfo{
+		Info:            *info,
+		ActivationEpoch: abi.ChainEpoch(seed),
+		DealWeight:      big.NewInt(int64(seed)),
+	}
+}
+
+// returns a unique SectorPreCommitInfo with each invocation with SectorNumber set to `sectorNo`.
+func (h *minerStateHarness) newSectorPreCommitInfo(sectorNo abi.SectorNumber) *miner.SectorPreCommitInfo {
+	seed := h.getSeed()
+	return &miner.SectorPreCommitInfo{
+		RegisteredProof: abi.RegisteredProof_StackedDRG32GiBPoSt,
+		SectorNumber:    sectorNo,
+		SealedCID:       h.cidGetter(),
+		SealRandEpoch:   abi.ChainEpoch(seed),
+		DealIDs:         nil,
+		Expiration:      abi.ChainEpoch(seed),
+	}
+}
+
+func (h *minerStateHarness) getSeed() uint64 {
+	defer func() { h.seed += 1 }()
+	return h.seed
+}
+
+// TODO consider allowing just the runtime store to be constructed, would need to change `AsStore` to operate on
+// runtime.Store instead  of the entire runtime.
+func constructStateHarness(t *testing.T, rt runtime.Runtime, periodBoundary abi.ChainEpoch) *minerStateHarness {
+	// store init
+	emptyMap, err := adt.MakeEmptyMap(adt.AsStore(rt)).Root()
+	require.NoError(t, err)
+
+	emptyArray, err := adt.MakeEmptyArray(adt.AsStore(rt)).Root()
+	require.NoError(t, err)
+
+	emptyDeadlines := miner.ConstructDeadlines()
+	emptyDeadlinesCid := rt.Store().Put(emptyDeadlines)
+
+	// state field init
+	owner := tutils.NewBLSAddr(t, 1)
+	worker := tutils.NewBLSAddr(t, 2)
+	state := miner.ConstructState(emptyArray, emptyMap, emptyDeadlinesCid, owner, worker, "peer", SectorSize, periodBoundary)
+
+	// assert NewSectors bitfield was constructed correctly (empty)
+	newSectorsCount, err := state.NewSectors.Count()
+	assert.NoError(t, err)
+	assert.Equal(t, uint64(0), newSectorsCount)
+
+	return &minerStateHarness{s: state, t: t, cidGetter: tutils.NewCidForTestGetter(), seed: 0}
+}
+
+func newBitFieldFromSectorNos(sectorNos ...abi.SectorNumber) *abi.BitField {
```

Comment:
Use `bitfield.NewFromSet`. You'll notice that some of the state methods use `uint64` in place of abi.SectorNumber to ease this.